src/lib/components/ng-http-loader.component.ts

```diff
@@ -111,7 +111,7 @@ export class NgHttpLoaderComponent implements OnDestroy, OnInit {
         if (showSpinner) {
             this.visibleUntil = Date.now() + this.minDuration;
         }
-        this.isSpinnerVisible = showSpinner;
+        this.isSpinnerVisible$ = of(showSpinner);
```

Comment:
I looked through the file and find an issue that actually confused me. Please take a look at my edit https://gist.github.com/byblik/0be812b2be750d1088d2dfe5af1ec416

Reply:
That won't work, we must keep subscriptions as is.

Reply:
Why?

Reply:
Because you can't remove the handleSpinnerVisibility method, which is responsible of the show/hide logic. Or I didn't not understood your gist.

Reply:
Actually you manually control the isSpinnerVisible variable though it can be perfectly done via observable stream itself.  That's the point. Your code works seems a bit redundant as you take the value from stream yourself and manually pass it to async further. Looks meaningless.

Reply:
oh, yes of course. That's just to avoid having too many logic in the stream itself. There's just no cost in adding a function here IMO.

Reply:
This is commendable to separate concerns but actually handleSpinnerVisibility is named wrong and placed wrong. Look, actually what this method does is changes timeout after which the spinner should be hidden (1 concern) and transforms visibility flag (2 concern) which is redundant itself.
You may add pipe(tap(updateVisibilityExpirationTime)), where updateVisibilityExpirationTime (or something like that) is extracted method with code
```ts
if (showSpinner) {
    this.visibleUntil = Date.now() + this.minDuration;
}
```

Reply:
You are plain right, doing this change now ! Thanks for feedback